include/bamit/sample_functions.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <algorithm> // For std::sort
+#include <random> // For std::random_device
+#include <numeric> // For std::reduce
+#include <cmath> // For std::sqrt and std::pow
+
+#include <bamit/Record.hpp>
+#include <bamit/IntervalNode.hpp>
+
+#include <seqan3/io/sam_file/input.hpp>
+#include <seqan3/core/debug_stream.hpp>
+
+namespace bamit
+{
+
+/*! Contains statistics which can be calculated over a sample. */
+struct EstimationResult
+{
+    double mean{0}, median{0}, mode{0}, sd{0}, variance{0};
+
+    /*! Print out the statistics stored within. */
+    void print()
+    {
+        seqan3::debug_stream << "Mean: " << mean
+                             << "\nMedian: " << median
+                             << "\nMode: " << mode
+                             << "\nSD: " << sd
+                             << "\nVariance: " << variance << "\n";
+    }
+};
+
+/*!
+   \brief A function to sample read depth from an alignment file, given the input file, index, and number of positions.
+   \param input_file The input alignment file in sam/bam format.
+   \param bamit_index The vector of indices for each chromosome.
+   \param sample_value The number of positions to sample.
+
```

Comment:
If I understood it corrrectly (maybe add this as details):
\details This functions takes the read depths from sample_value equally distributed positions and gives the read depth of the file as a Gaussian distribution with mean, median, mode, sd and variance stored in a EstimationResult struct.